addons/survey/tests/test_survey_results.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.survey.controllers.main import Survey
+from odoo.addons.survey.tests import common
+from odoo.addons.website.tools import MockRequest
+
+
+class TestSurveyResults(common.TestSurveyResultsCommon):
+    """ Check the results and the performance of the different filters combinations.
+    The filters can be combined but their query count doesn't add up if their
+    related question answers data are stored in the same model.
+    """
+
+    def setUp(self):
+        super(TestSurveyResults, self).setUp()
+        self.SurveyController = Survey()
+
+    def test_get_filters_from_post(self):
+        """ Check that the filters are correctly retrieved from the post. """
+        post = {'filters': 'A,14,101|A,0,58|L,0,2'}
```

Comment:
Maybe we could put multiple filters on the matrix for completeness?

It's probably already covered below but so is most of the code of "_get_filters_from_post", so I guess it does not hurt. More simple to see that the extracted filter does not match if that test fails that retro-engineering why the results do not match.